R/sits_kohonen.R

```diff
@@ -1,92 +1,770 @@
+            positon_to_add<-which(Label_table==no_cluster[sv])
+
+            #trasform in df.array to add a column
+            cf<-as.data.frame.array(confusion.matrix)
+
+            #add a column in same position of line
+            cf<-add_column(cf, d = 0, .after =positon_to_add-1 )
+
+            #rename de column to name of cluster
+            names(cf)[positon_to_add]<-no_cluster[sv]
+            confusion.matrix<-as.table(as.matrix(cf))
+        }
+    }
+
+    #Add the total number of samples in table
+    confusion.matrix.tb <- addmargins(confusion.matrix, FUN = list(Total = sum), quiet = TRUE)
+
+    #get dimensions (rows and col)
+    #represents the original classes of samples
+    dim_row <-dim(confusion.matrix.tb)[1]
+
+    #represents clusters
+    dim_col <-dim(confusion.matrix.tb)[2]
+
+    #get the names of classes (original labels from samples)
+    Label_table<-rownames(confusion.matrix.tb)[1:dim_row-1]
+
+    Mix_class <- dplyr::tibble()
+    d = 1
+    for (d in 1:(dim_row - 1))
+    {
+        #sum the samples of label "d" by cluster.
+        #each column represents the cluster where the sample was allocated
+        current_row <- confusion.matrix.tb[d, 1:dim_col-1]
+        current_col <- confusion.matrix.tb[1:dim_row-1,d]
+
+        #get the value total
+        current_row_Total <- confusion.matrix.tb[d, dim_col]
+        current_col_Total <- confusion.matrix.tb[dim_row, d]
+
+        current_class_ambiguity <-
+            tibble::as.tibble(list(
+                id_class= as.integer(d),
+                cluster= Label_table[d],
+                original_class = names(current_col),
+                mixture_percentage = as.numeric((current_col / current_col_Total) *
+                                                    100)
+            ))
+
+        #sort the table in decending order
+        current_class_ambiguity<- dplyr::arrange(current_class_ambiguity,
+                                                 desc(current_class_ambiguity$mixture_percentage))
+
+        #remove lines that mix_percentege is zero
+        current_class_ambiguity <-
+            dplyr::filter(current_class_ambiguity,
+                          current_class_ambiguity$mixture_percentage > 0)
+
+        Mix_class<-rbind(Mix_class,current_class_ambiguity)
+    }
+
+    info_confusion_matrix <-
+        caret::confusionMatrix(confusion.matrix.tb[1:dim_row-1,1:dim_col-1])
+
+    metrics_by_cluster <-  structure(list(
+        mixture_cluster= Mix_class,
+        confusion_matrix = info_confusion_matrix),
+        class = "sits")
+
+    return (metrics_by_cluster)
+
+}
+
+#' @title Configure pallete
+#' @name .sits_kohonen_pallete
+#' @author Lorena Santos, \email{lorena.santos@@inpe.br}
+#'
+#' @description configure a pallete with 21 colors.
+#'
+#' @return rreturn a pallete of colors
 
-    # create a "cluster" column
-    data.tb$cluster <- as.integer(kohonen.obj$unit.classif)
+.sits_kohonen_pallete <- function()
```

Comment:
Is there no color pallete with such colors already implemented in R? R has already a function called `rainbow()`.